tiledb/sm/cpp_api/query.h

```diff
@@ -395,6 +394,120 @@ class Query {
     return elements;
   }
 
+  /**
+   * Adds a 1D range along a subarray dimension, in the form
+   * (start, end, stride). The datatype of the range
+   * must be the same as the subarray domain type.
+   *
+   * **Example:**
+   *
+   * @code{.cpp}
+   * // Set a 1D range on dimension 0, assuming the domain type is int64.
+   * int64_t start = 10;
+   * int64_t end = 20;
+   * // Stride is optional
+   * subarray.add_range(0, start, end);
+   * @endcode
+   *
+   * @param dim_idx The index of the dimension to add the range to.
+   * @param range The range to add.
+   * @return Reference to this Subarray
+   */
+  Query& add_range(
```

Comment:
should this method be templated to avoid raw void pointers?

Reply:
Will address.